.github/workflows/ci.yml

```diff
@@ -29,92 +28,69 @@ jobs:
       - name: typos
         uses: crate-ci/typos@master
 
-      - name: setup uv
-        uses: astral-sh/setup-uv@v4
-        with:
-          enable-cache: true
-          python-version: "3.13"
-
-      - name: install
-        run: uv sync
-
-      - name: repo-review
-        run: uv run repo-review .
-
-      - name: mdformat
-        run: uv run mdformat --check
-
-      - name: ruff
-        run: |
-          uv run ruff check --output-format=github
-          uv run ruff format --check
-
-  typetest:
+  generate-matrix:
+    name: Generate Test Matrix
     timeout-minutes: 5
     runs-on: ubuntu-latest
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
+      - name: Install uv
         uses: astral-sh/setup-uv@v4
-        with:
-          enable-cache: true
-          python-version: "3.13"
 
-      - name: install
-        run: uv sync --no-editable
+      - name: Install dependencies
+        run: |
+          uv sync --frozen
+          uv pip install packaging
 
-      - name: basedpyright
-        run: >
-          uv run --no-sync
-          basedpyright
+      - name: Generate Matrix
+        id: set-matrix
+        shell: bash
+        run: |
+          matrix=$(uv run scripts/generate_matrix.py | jq -c)
+          echo "matrix=$matrix" >> $GITHUB_OUTPUT
 
-      - name: basedmypy
-        run: >
-          uv run --no-sync
-          mypy --config-file=pyproject.toml .
-
-      - name: stubtest
-        run: >
-          uv run --no-sync
-          stubtest
-          --mypy-config-file=pyproject.toml
-          --allowlist=.mypyignore
-          --ignore-unused-allowlist
-          scipy
-
-  typetest-oldest-supported-numpy:
-    timeout-minutes: 5
+  typetest:
+
+    needs: generate-matrix
+    timeout-minutes: 20
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.generate-matrix.outputs.matrix) }}
     steps:
       - uses: actions/checkout@v4
 
-      - name: setup uv
+      - name: Install uv
         uses: astral-sh/setup-uv@v4
         with:
           enable-cache: true
-          python-version: "3.10"
+          python-version: ${{ matrix.python }}
```

Comment:
doesn't this remove the need for the explicit `--python=...` later on?

Reply:
Maybe? I'll test